target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
Good job so far! Watchdog support would be a nice addition but it's more nice to have.

I can see the following problems:

> ```
> [    0.876047] lantiq nor flash device: 02000000 at 10000000
> [    0.881766] ltq-nor 10000000.nor-boot: probing failed
> ```

If the NOR flash chip can not be probed, no partition can be mounted.

The next point is that all log messages printed during early boot are on a single line:

> ```
> [    0.000000] Memory: 122972K/131072K available (4019K kernel code, 168K rwdata, 1172K rodata, 1216K init, 222K bss, 8100K reserved, 0K cma[    0.858966] console [ttyLTQ0] enabledat MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,ascHat, Inc.
> ```

It seams to me linebreaks  don't work  in early bootconsole and the same line gets overwritten over and over again...

I've no idea what the bootmonitor does (or does not do in compare to u-boot) that you are facing the issues. But I had similar issues with brnboot before, which was the reason why I added [lantiq,phy-clk-src ](https://github.com/openwrt/openwrt/commit/f9151d154f7a6f0523dd09f348178bbe788b8ef0).

Due to no public available datasheets, debugging the issues will be for sure a hard job. While I debugged my phy-clk-src issue back in the days, I dumped all lantiq specific registers I found in the u-boot source during kernel start and compared the values between a boot via brnboot and u-boot. But maybe at least the flash issue is something similar to https://github.com/openwrt/openwrt/commit/c149f7360578de58131478194ba233f1bbaba661.

As long as the flash issue persist, it doesn't make sense to me to fix the remaining stuff like:

- adjusting the partition table (uboot is gone)
- add support for the boss header + mtdsplitter
- try to load a not gzip compressed image from flash OR compress the already lzma compressed kernel/image with gzip to make bootmonitor happy